### PR TITLE
Improve diagnostic messages for IDE0060 (remove unused parameter)

### LINKED ISSUE #38050: IDE0060 doesn't consider nameof to be a use
**Version Used**: 
3.3.0-beta3-19413-06+ac06df1bffb7b7fd0e5bf63cc91921af76b21e03

**Steps to Reproduce**:
```C#
private static void ValidateStartIndexAndLength_Throw(int startIndex, int length) =>
    throw new ArgumentOutOfRangeException(startIndex < 0 ? nameof(startIndex) : nameof(length));
```

**Expected Behavior**:
No warnings.

**Actual Behavior**:
IDE0060 triggers "Remove unused parameter 'length'"

Is that by-design?

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #38050

Improve IDE0060 messages for parameter which has non-zero references, but still can be removed as the initial parameter value is never used.

### CODE REVIEW

**Comment by mavasani** [comment]
Ping for reviews. Trivial PR changing diagnostic messages for clarity.

**Comment by mavasani** [comment]
Ping @dotnet/roslyn-ide for review.

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs [hunk 1]
```diff
@@ -733,7 +733,8 @@ static bool IsRemovableAssignmentValueWithoutSideEffects(IOperation assignmentVa
 
                         if (!isUsed)
                         {
-                            _symbolStartAnalyzer._unusedParameters[parameter] = isSymbolRead;
+                            _symbolStartAnalyzer._unusedParameters[parameter] = isSymbolRead ||
+                                _referencedParameters.ContainsKey(parameter);
```
Comment (genlu) [inline src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs]: maybe add some comments here?

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs [hunk 2]
```diff
@@ -733,7 +733,8 @@ static bool IsRemovableAssignmentValueWithoutSideEffects(IOperation assignmentVa
 
                         if (!isUsed)
                         {
-                            _symbolStartAnalyzer._unusedParameters[parameter] = isSymbolRead;
+                            _symbolStartAnalyzer._unusedParameters[parameter] = isSymbolRead ||
```
Comment (mavasani) [inline src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs]:
```suggestion
                            // Mark if the symbol's value is read or the symbol is referenced to ensure appropriate diagnostic message is given.
                            _symbolStartAnalyzer._unusedParameters[parameter] = isSymbolRead ||
```

**Pull request merged**